CODE-GUIDELINES.md

```diff
@@ -0,0 +1,41 @@
+# Guidelines for Podman Desktop Code
+
+## Production code
+
+## Unit tests code
+
+### Use `vi.mocked`, not a generic `myFunctionMock`
+
+If you define a mock with `const myFunctionMock = vi.fn();` its type is `Mock<Procedure>`, which is a generic type.
+
+For example, do not write this, or Typescript won't be able to detect that you passed an object instead of a string to `mockResolvedValue`:
+
+```ts
+const windowMethodMock = vi.fn();
+
+Object.defineProperty(global, 'window', {
+  value: {
+    windowMethod:windowMethodMock,
+  },
+  writable: true,
```

Comment:
Is it required ?

Reply:
for this example, at least, no. I'm removing it